common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockHandler.java

```diff
@@ -64,6 +75,10 @@ public ExternalShuffleBlockHandler(TransportConf conf, File registeredExecutorFi
   public ExternalShuffleBlockHandler(
       OneForOneStreamManager streamManager,
       ExternalShuffleBlockResolver blockManager) {
+    this.metrics = new ShuffleMetrics();
+    this.timeDelayForOpenBlockRequest = metrics.timeDelayForOpenBlockRequest;
+    this.timeDelayForRegisterExecutorRequest = metrics.timeDelayForRegisterExecutorRequest;
```

Comment:
It's a little confusing how this metric is duplicated as a class member. Would it work to just reference it through `metrics`?